filebeat/values.yaml

```diff
@@ -21,7 +25,11 @@ filebeatConfig:
 # Extra environment variables to append to the DaemonSet pod spec.
 # This will be appended to the current 'env:' key. You can use any of the kubernetes env
 # syntax here
-extraEnvs: []
+extraEnvs: 
```

Comment:
`NODE_NAME` is already set in https://github.com/elastic/helm-charts/blob/917187d3c6c30099aaf81681d8df7592006a25b3/filebeat/templates/daemonset.yaml#L124-L127

Reply:
The /var/log/containers does not work actually as logs are not in that path in any k8s in any cloud be it GKE/EKS/AKS and on installing with the default values, the logs error out in kibana .. The Logs exist in /var/lib/docker/containers/*/*.log with or without autodiscover

Reply:
Of course we can override it via inline set but it may help to provide a working filebeat values.yaml as the default

Reply:
`/var/log/containers` is the standard log path for kubernetes pods logs and I tested it again with GKE 1.14 and 1.16:

```
➜  ~ helm install filebeat elastic/filebeat
NAME: filebeat
LAST DEPLOYED: Thu Jul  9 08:36:46 2020
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
1. Watch all containers come up.
  $ kubectl get pods --namespace=default -l app=filebeat-filebeat -w
➜  ~ kubectl exec filebeat-filebeat-7jnsq -- bash
[root@filebeat-filebeat-7jnsq filebeat]# ls -l /var/log/containers/
total 40
lrwxrwxrwx 1 root root  97 Jul  9 06:37 filebeat-filebeat-7jnsq_default_filebeat-46ec2e693a5ff60dcf71437ec2d25d9e02ce036140d1d4c2d68f8170b900afdc.log -> /var/log/pods/default_filebeat-filebeat-7jnsq_90588f55-c1ae-11ea-ac2a-42010a84002c/filebeat/0.log
lrwxrwxrwx 1 root root 105 Jul  9 06:36 fluentd-gcp-v3.1.1-5hr8l_kube-system_fluentd-gcp-3f44021a5f3ea1a089b0cba5faa0d950c26a5fdd46bea032a1cbcdc00fb09f0f.log -> /var/log/pods/kube-system_fluentd-gcp-v3.1.1-5hr8l_8ef16f80-c1ae-11ea-ac2a-42010a84002c/fluentd-gcp/0.log
lrwxrwxrwx 1 root root 119 Jul  9 06:36 fluentd-gcp-v3.1.1-5hr8l_kube-system_prometheus-to-sd-exporter-1b5c13aefbe81a9bbad65036ffe7c26b25fb10d86c65f4d320aa92686c88e116.log -> /var/log/pods/kube-system_fluentd-gcp-v3.1.1-5hr8l_8ef16f80-c1ae-11ea-ac2a-42010a84002c/prometheus-to-sd-exporter/0.log
lrwxrwxrwx 1 root root 102 Jul  9 06:36 kube-dns-5995c95f64-5bng6_kube-system_dnsmasq-a07a1b526474aceba81ae9707ae79a72cccda6596c74343b0d92c2e780956c16.log -> /var/log/pods/kube-system_kube-dns-5995c95f64-5bng6_737f767f-c1ae-11ea-ac2a-42010a84002c/dnsmasq/0.log
lrwxrwxrwx 1 root root 102 Jul  9 06:36 kube-dns-5995c95f64-5bng6_kube-system_kubedns-d46e74cd4e54bfcbf552a7d6a24fe5542da21d755ec96972eaf29585c015a17b.log -> /var/log/pods/kube-system_kube-dns-5995c95f64-5bng6_737f767f-c1ae-11ea-ac2a-42010a84002c/kubedns/0.log
lrwxrwxrwx 1 root root 111 Jul  9 06:36 kube-dns-5995c95f64-5bng6_kube-system_prometheus-to-sd-9e3ee467856a458a21b70301dfb404851b02e640eb6281915c876d43b4cc4b93.log -> /var/log/pods/kube-system_kube-dns-5995c95f64-5bng6_737f767f-c1ae-11ea-ac2a-42010a84002c/prometheus-to-sd/0.log
lrwxrwxrwx 1 root root 102 Jul  9 06:36 kube-dns-5995c95f64-5bng6_kube-system_sidecar-ffa584156e8fa85438977da523132fe75858131e0d429b7cdd11237f37c459b5.log -> /var/log/pods/kube-system_kube-dns-5995c95f64-5bng6_737f767f-c1ae-11ea-ac2a-42010a84002c/sidecar/0.log
lrwxrwxrwx 1 root root 128 Jul  9 06:36 kube-proxy-gke-test-jmlrt-default-pool-bdef7ba5-cc84_kube-system_kube-proxy-d5548f3a25a0058db1a97af19c651e38bef457768d2dce2798d6e8038b3d372f.log -> /var/log/pods/kube-system_kube-proxy-gke-test-jmlrt-default-pool-bdef7ba5-cc84_eba85cbee6b0c94144a742cddb7c7148/kube-proxy/0.log
lrwxrwxrwx 1 root root 108 Jul  9 06:36 prometheus-to-sd-7k2g5_kube-system_prometheus-to-sd-7cccae275b10624636bfc473434edfc9031aeb3353e15d4e6f53ee8bc08b6bac.log -> /var/log/pods/kube-system_prometheus-to-sd-7k2g5_76c9254c-c1ae-11ea-ac2a-42010a84002c/prometheus-to-sd/0.log
lrwxrwxrwx 1 root root 118 Jul  9 06:36 prometheus-to-sd-7k2g5_kube-system_prometheus-to-sd-new-model-b9873fffde0410dc2a289ca61d57ac94ffc5671b67dca0f8cc4c78918c6510c5.log -> **/var/log/pods/kube-system_prometheus-to-sd-7k2g5_76c9254c-c1ae-11ea-ac2a-42010a84002c/prometheus-to-sd-new-model/0.log**
```

We don't have the workload to deeply test every use cases and every K8S distribution, so we regularly discover new bugs and we are relying on GitHub Issues/PR for additionnal feedback.

However, the recommended Filebeat configuration for K8S is deeply tested by Filebeat developers, we use internally these charts with tenth of GKE clusters in production, we have tests which are running every day in our CI with GKE clusters and ensure that a basic scenario like Filebeat with default values can send pod logs to Elasticsearch and finally we have feedbacks from tenth of external users which are using these charts in production since more than a year.

So rather than thinking that this chart doesn't work with default config and has never been tested and submit a PR to change the default configuration for something which may work in your use case but may not in the tenth of other use cases already running in production, I would just assume that maybe there is something which differ from the configuration which are tested and known to work in your environment and request help in https://discuss.elastic.co or in an issue with as much details as you have about your specific environment, config and errors.